JS-Algorithms/challenges.js

```diff
@@ -95,7 +133,19 @@ Since 10! === 3628800 and you sum 3 + 6 + 2 + 8 + 8 + 0 + 0
 ***** */
 
 const digitSum = (n) => {
-  // YOUR CODE HERE...
+  let factorialOfN = BigInt(1);
+
+  for (let i = BigInt(n); i > 1; i--) {
+    factorialOfN *= i;
+  }
+
+  const digitsInfactorialOfN = [...factorialOfN.toString()].map(
+    (digit) => Number(digit)
+  );
+
+  const sumOfDigits = digitsInfactorialOfN.reduce((acc, num) => acc + num, 0);
+
+  return sumOfDigits;
```

Comment:
Clever solution you can apply the number cast directly to the reduce function instead of separate it in two steps though 

` const sumOfDigits = [...factorialOfN.toString()].reduce((acc, digit) => acc + Number(digit), 0);
`